### PR TITLE
Fix `anydata` to `json` cast

### LINKED ISSUE #36972: Invalid `map<anydata>` to `json` cast doesn't panic
**Description:**
$title.

**Steps to reproduce:**
```ballerina
import ballerina/io;

public function main() {
    record {|
        anydata body;
    |} badRequest = { 
        body: { 
            code: "ERROR_CODE", 
            details: "ERROR_DETAILS"
        }
    };
    
    io:println(badRequest?.body is json); // false

    // So the following cast should panic, but doesn't atm.
    json j = <json> badRequest?.body;
}
```

**Affected Versions:**
2201.2.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #36972

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/expressions/typecast/type_cast_expr.bal [hunk 1]
```diff
@@ -265,6 +265,15 @@ function testJsonCastNegative() {
...
     assertTypeCastFailureWithMessage(trap <json> a, "incompatible types: 'lang.xml:Text' cannot be cast to 'json'");
+
+    anydata val1 = {"a":1};
+    json|error jval1 = trap <json> val1;
+    test:assertTrue(jval1 is error);
+    if (jval1 is error) {
+        test:assertEquals("{ballerina}TypeCastError", jval1.message());
+        test:assertEquals("incompatible types: 'map<anydata>' cannot be cast to 'json'",
+        <string> checkpanic jval1.detail()["message"]);
+    }
```
Comment (HindujaB) [inline tests/jballerina-unit-test/src/test/resources/test-src/expressions/typecast/type_cast_expr.bal:276]: Shall we add a `record` with the `anydata` field test case similar to the one mentioned in the issue?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCastGen.java [hunk 2]
```diff
@@ -1319,15 +1319,18 @@ private void generateCheckCastToAnyData(MethodVisitor mv, BType type) {
...
+            case TypeTags.ANY:
+            case TypeTags.ANYDATA:
+            case TypeTags.UNION:
+            case TypeTags.INTERSECTION:
+            case TypeTags.READONLY:
+            case TypeTags.MAP:
+                checkCast(mv, symbolTable.jsonType);
+                break;
+            default:
+                // if value types, then ad box instruction
```
Comment (HindujaB) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCastGen.java:1332]:
```suggestion
                // for value types, then add box instruction
```

**Review by HindujaB — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]

**Review by warunalakshitha — COMMENTED** [review]
Lets send this to master as well

**Pull request merged**